commit 52a004d146bad5a390435bc6d7716c4c96639b23 (refs/changes/36/5036/2)
Author: Josh Wilsdon <josh@wilsdon.ca>
Date:   2018-11-07T11:48:29-08:00 (11 months ago)
    
    TRITON-940 Add metricPorts metadata to binder for cmon-agent
    Reviewed by: Dylan Yep <dyep49@gmail.com>
    Approved by: Dylan Yep <dyep49@gmail.com>

diff --git a/boot/setup.sh b/boot/setup.sh
index 85ab7de..75cbbdc 100755
--- a/boot/setup.sh
+++ b/boot/setup.sh
@@ -156,6 +156,9 @@ else # FLAVOR == "sdc"
     sdc_log_rotation_add zookeeper /var/log/zookeeper/zookeeper.log 1g
     sdc_log_rotation_setup_end
 
+    # Add metadata for cmon-agent discovery
+    mdata-put metricPorts 1053
+
     # All done, run boilerplate end-of-setup
     sdc_setup_complete
 
diff --git a/lib/server.js b/lib/server.js
index 078b4d3..13c764b 100644
--- a/lib/server.js
+++ b/lib/server.js
@@ -31,7 +31,7 @@ dtp.enable();
 
 // Binder metric collectors
 var METRIC_REQUEST_COUNTER = 'binder_requests_completed';
-var METRIC_LATENCY_HISTOGRAM = 'binder_request_latency_ms';
+var METRIC_LATENCY_HISTOGRAM = 'binder_request_latency_seconds';
 var METRIC_SIZE_HISTOGRAM = 'binder_response_size_bytes';
 
 ///--- Helpers
@@ -526,7 +526,7 @@ function createServer(options) {
 
                 var labels = { type: query.type() };
                 query_counter.increment(labels);
-                latency_histogram.observe(lat, labels);
+                latency_histogram.observe(lat / 1000, labels);
                 size_histogram.observe(query.bytesSent, labels);
 
                 query._log[loglevel]({
diff --git a/package.json b/package.json
index 30dbcdf..8f844b0 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "binder",
   "description": "SmartDataService DNS Service",
-  "version": "1.2.0",
+  "version": "1.2.1",
   "author": "Joyent (joyent.com)",
   "license": "MPL-2.0",
   "private": true,
